rotkehlchen/tests/unit/test_assets.py

```diff
@@ -633,3 +639,43 @@ def test_symbol_or_name(database):
     assert Asset('xyz').symbol_or_name() == 'custom name'
     with pytest.raises(UnknownAsset):
         Asset('i-dont-exist').symbol_or_name()
+
+
+def test_load_from_packaged_db(globaldb: GlobalDBHandler):
+    """
+    Test that when resolving assets from the packaged db,
+    there is no attempts to write into the pacakged db.
+    """
+    packaged_db_path = Path(__file__).resolve().parent.parent.parent / 'data' / 'global.db'
+    with TemporaryDirectory() as tmpdirname:
+        backup = shutil.copy(packaged_db_path, Path(tmpdirname, 'globaldb.backup'))
+        try:
+            # Delete bitcoin from the loaded db
+            globaldb.delete_asset_by_identifier(identifier='BTC')
+            packaged_db_patch = patch.object(
+                GlobalDBHandler,
+                'packaged_db_conn',
+                wraps=globaldb.packaged_db_conn,
+            )
+            packaged_db_conn = DBConnection(
+                path=packaged_db_path,
+                connection_type=DBConnectionType.GLOBAL,
+                sql_vm_instructions_cb=0,
+            )
+            with packaged_db_conn.write_ctx() as cursor:
+                # Drop the settings table in order to provoke insertion of tables when
+                # the initializaion of the packaged db will happen.
+                cursor.execute('DROP TABLE IF EXISTS settings')
+
+            hash_before = file_md5(packaged_db_path)
+            with packaged_db_patch as packaged_db_mock:
+                # Try to load the asset
+                CryptoAsset('BTC')
+                # Check that the packaged db was used
+                assert packaged_db_mock.call_count == 1
+
+            # Check that the packaged db was not modified
+            hash_after = file_md5(packaged_db_path)
+            assert hash_before == hash_after
+        finally:
+            shutil.copy(backup, packaged_db_path)
```

Comment:
why do you do this? Why wrap the entire thing in try/finally? And why at the finally copy back?

Reply:
Because to provoke the execution of the script that creates new tables, I delete settings table form the packaged db

Reply:
And in order to restore the packaged db after the db, I added this `finally:` block

Reply:
Okay then this entire test is very bad. Even in tests you must not write on the packaged DB. Mock it please. Also perhaps you can just see if what @yabirgb mentioned is true. Perhaps just an attempt to write is what breaks and testing the md5hash makes no sense.